packages/workflow/src/records/handler/correction/utils.ts

```diff
@@ -68,3 +71,50 @@ export async function sendNotification<T extends keyof PayloadMap>(
 
   return res
 }
+
+const replaceTmpURN = (resource: any, tmpURN: string, fullUrl: string) => {
+  if (Array.isArray(resource)) {
+    resource.forEach((item, index) => {
+      if (item === tmpURN) {
+        resource[index] = fullUrl
+      } else {
+        replaceTmpURN(item, tmpURN, fullUrl)
+      }
+    })
+  } else if (typeof resource === 'object' && resource !== null) {
+    for (const key in resource) {
+      if (resource[key] === tmpURN) {
+        resource[key] =
+          key === 'reference'
+            ? fullUrl.split('/').slice(2, 4).join('/') // /fhir/Patient/xyz/_history/abc => Patient/xyz
```

Comment:
perhaps we can use these utility functions: https://github.com/opencrvs/opencrvs-core/blob/a553658245b0d34a00f262f0f2bb326b79366ba5/packages/commons/src/fhir/index.ts#L60-L76